app/models/smart_proxy.rb

```diff
@@ -89,15 +89,15 @@ def has_feature?(feature_name)
   end
 
   def capabilities(feature)
-    smart_proxy_features.find_by(:feature_id => Feature.find_by(:name => feature)).try(:capabilities)
+    smart_proxy_features.find { |spf| spf.feature_name.to_s == feature.to_s }.try(:capabilities)
```

Comment:
Maybe we can add some `includes` statements here and in `settings` to make it a bit more efficient:

```suggestion
    smart_proxy_features.includes(:feature).find { |spf| spf.feature_name.to_s == feature.to_s }.try(:capabilities)
```

Reply:
:+1:

Reply:
actually yes, this doesn't seem to work with unsaved objects:

```
>  SmartProxy.new(:smart_proxy_features => [SmartProxyFeature.new(:feature_id => 1, :capabilities => ['foo'])]).smart_proxy_features
=> [#<SmartProxyFeature:0x00000000128e03c0 smart_proxy_id: nil, feature_id: 1, id: nil, capabilities: ["foo"], settings: {}>]

> SmartProxy.new(:smart_proxy_features => [SmartProxyFeature.new(:feature_id => 1, :capabilities => ['foo'])]).smart_proxy_features.includes(:feature)
=> []


```

Reply:
So actually reverting back to this would cause only 1 query per object:
```
    smart_proxy_features.find { |spf| spf.feature_id == Feature.find_by(:name => feature) }.try(:capabilities)
```

thoughts?